src/snowflake/snowpark/_internal/analyzer/select_statement.py

```diff
@@ -880,7 +880,7 @@ def __init__(
         self._projection_in_str = None
         self._query_params = None
         self.expr_to_alias.update(self.from_.expr_to_alias)
-        self.df_aliased_col_name_to_real_col_name.update(
+        self.df_aliased_col_name_to_real_col_name = deepcopy(
```

Comment:
deepcopy is not equivalent to update.

if there is a "key" in `self.df_aliased_col_name_to_real_col_name` but not in `self.from_.df_aliased_col_name_to_real_col_name`.

the previous update will keep the "key", but deepcopy will discard the "key".

will this cause issue?

Reply:
I agree. However this is the `__init__` method. At this point, `self.df_aliased_col_name_to_real_col_name` should be empty generated by `Selectable` init https://github.com/snowflakedb/snowpark-python/blob/1a5121934cddc372e54b8d7e734730dc003331ce/src/snowflake/snowpark/_internal/analyzer/select_statement.py#L265